api/utils.ts

```diff
@@ -0,0 +1,36 @@
+import { getExperimentsAuthInfo } from '@/utils/auth'
+
+import UnauthorizedError from './UnauthorizedError'
+
+const DEVELOPMENT_API_URL_ROOT = 'https://virtserver.swaggerhub.com/yanir/experiments/0.1.0'
+const PRODUCTION_API_URL_ROOT = 'https://public-api.wordpress.com/wpcom/v2/experiments/0.1.0'
+
+/**
+ * Makes a request to the Experiment Platform's API with any necessary
+ * authorization information, parses the response as JSON, and returns the parsed
+ * response.
+ *
+ * Note: Be sure to handle any errors that may be thrown.
+ *
+ * @throws UnauthorizedError
+ */
+async function fetchApi(method: string, path: string, body: BodyInit | null = null) {
```

Comment:
It may be a good idea to use const enums instead of a string, or even just a whitelist possible set of strings to get some of that intellisense magic:

```suggestion
export const enum AvailableEnpoints { // or a better name?
  Experiments = '/experiments'
}

async function fetchApi(method: string, path: AvailableEndpoints, body: BodyInit | null = null) {
```

Or

```suggestion
type ExperimentEndpoints = "/experiments" | "/metrics"

async function fetchApi(method: string, path: ExperimentEndpoints, body: BodyInit | null = null) {
```

Reply:
I like.

Reply:
This won't work well once we have dynamic strings. I think it overcomplicates things for this PR.

Reply:
Also, keep in mind that this function is a helper that's internal to the API client. Outside the client, we will call specific functions. Let's not overdo type safety unnecessarily. :slightly_smiling_face: